velox/common/encode/Base64.cpp

```diff
@@ -320,29 +322,35 @@ void Base64::decode(
     const std::pair<const char*, int32_t>& payload,
     std::string& decodedOutput) {
   size_t inputSize = payload.second;
-  decodedOutput.resize(calculateDecodedSize(payload.first, inputSize));
-  decode(payload.first, inputSize, decodedOutput.data(), decodedOutput.size());
+  size_t decodedSize;
+  (void)calculateDecodedSize(payload.first, inputSize, decodedSize);
+  decodedOutput.resize(decodedSize);
+  (void)decode(
+      payload.first, inputSize, decodedOutput.data(), decodedOutput.size());
 }
 
 // static
-void Base64::decode(const char* input, size_t size, char* output) {
-  size_t expectedOutputSize = size / 4 * 3;
-  Base64::decode(input, size, output, expectedOutputSize);
+void Base64::decode(const char* input, size_t inputSize, char* outputBuffer) {
+  size_t outputSize;
+  (void)calculateDecodedSize(input, inputSize, outputSize);
+  (void)decode(input, inputSize, outputBuffer, outputSize);
 }
 
 // static
 uint8_t Base64::base64ReverseLookup(
     char encodedChar,
-    const Base64::ReverseIndex& reverseIndex) {
-  auto reverseLookupValue = reverseIndex[(uint8_t)encodedChar];
+    const ReverseIndex& reverseIndex,
+    Status& status) {
+  auto reverseLookupValue = reverseIndex[static_cast<uint8_t>(encodedChar)];
   if (reverseLookupValue >= 0x40) {
-    VELOX_USER_FAIL("decode() - invalid input string: invalid characters");
+    status = Status::UserError(
+        "decode() - invalid input string: invalid characters");
```

Comment:
We should probably print the invalid character here.

Reply:
Updated the code and added the testcases as well.